modules/ckanext-ytp_main/ckanext/ytp/validators.py

```diff
@@ -459,3 +459,21 @@ def check_deprecation(key, data, errors, context):
 
     deprecation = check_package_deprecation(data.get(('valid_till',)))
     data[key] = deprecation
+
+
+@scheming_validator
+def admin_only_field(field, schema):
+    from ckan.lib.navl.dictization_functions import flatten_dict
+    from ckan.logic import get_action
+
+    def validator(key, data, errors, context):
+
+        if 'package' not in context:
+            return
+
+        data_dict = flatten_dict(get_action('package_show')(context, {'id': context['package'].id}))
+
+        if not authz.is_sysadmin(context['user']):
+            data[key] = data_dict[key]
```

Comment:
What does this do? It doesn't modify the data['key'] (so the field) if the user is a sysadmin? The update here is to keep the old value?

Reply:
Yes. If the user is not an admin, we revert whatever change was made.

Reply:
Ah yes